File: SecurityWithIOT/SecurityWithIOT.SPA/src/app/_models/Address.ts
import { Country } from './Country';
import { City } from './City';
import { District } from './District';
import { User } from './User';

export interface Address {
    id: number;
    description: string;
    country: Country;
    city: City;
    district: District;
    user: User;
}

File: SecurityWithIOT/SecurityWithIOT.API/Model/Entities/Address.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SecurityWithIOT.API.Model
{
    [Table("Address")]
    public  class Address : BaseEntity
    {
        public string Description {get;set;}

        public int CityId {get;set;}
        public virtual City City {get;set;}

        public int DistrictId {get;set;}
        public virtual District District {get;set;}

        public int CountryId {get;set;}
        public virtual Country Country {get;set;}
        public int UserId {get;set;}
        public virtual User User {get;set;}
    }
}